engine/jobmaster/dm/checkpoint/agent_test.go

```diff
@@ -269,3 +274,38 @@ func TestIsFresh(t *testing.T) {
 	require.Error(t, err)
 	require.False(t, isFresh)
 }
+
+func TestFetchTableStmt(t *testing.T) {
+	var (
+		source   = "source"
+		database = "db"
+		table    = "table"
+	)
+	agent := NewAgentImpl("job_id", log.L())
+	db, mockDB, err := conn.InitMockDBNotClose()
+	require.NoError(t, err)
+	defer db.Close()
+
+	p := parser.New()
+	tracker, err := schema.NewTestTracker(context.Background(), "test-tracker", nil, dlog.L())
+	require.NoError(t, err)
+	stmt := "CREATE DATABASE `db`"
+	ret, err := p.ParseOneStmt(stmt, "", "")
+	require.NoError(t, err)
+	require.NoError(t, tracker.Exec(context.Background(), database, ret))
+
+	stmt = "CREATE TABLE `table` ( `id` int(11) DEFAULT NULL) ENGINE=InnoDB DEFAULT CHARSET=latin1"
+	ret, err = p.ParseOneStmt(stmt, "", "")
+	require.NoError(t, err)
+	require.NoError(t, tracker.Exec(context.Background(), database, ret))
+
+	tbInfo, err := tracker.GetTableInfo(&filter.Table{Schema: database, Name: table})
+	require.NoError(t, err)
+	bs, err := json.Marshal(tbInfo)
+	require.NoError(t, err)
+
+	mockDB.ExpectQuery(regexp.QuoteMeta("SELECT table_info FROM `meta`.`job_id_syncer_checkpoint` WHERE id = ? AND cp_schema = ? AND cp_table = ?")).WithArgs(source, database, table).WillReturnRows(sqlmock.NewRows([]string{"table_info"}).AddRow(bs))
+	tbStmt, err := agent.FetchTableStmt(context.Background(), "job_id", &config.JobCfg{MetaSchema: "meta"}, metadata.SourceTable{Source: source, Schema: database, Table: table})
+	require.NoError(t, err)
+	require.Equal(t, "CREATE TABLE `tbl`(`id` INT(11)) CHARSET LATIN1 COLLATE LATIN1_BIN", tbStmt)
```

Comment:
if this content is displayed to user, we should make sure it does not missing noticeable elements (like missing comments?) which makes user confused about the displayed table is at which history version.

And I noticed that the Table.Restore in Table.String has comment "for debug use only", maybe it will not generate a legal CREATE TABLE statement in some case, I prefer we use `ConstructResultOfShowCreateTable`